tests/test_questionnaire_validator.py

```diff
@@ -406,3 +406,21 @@ def test_invalid_calculated_or_grand_calculated_summary_id_in_value_source():
     validator.validate()
 
     assert validator.errors == expected_errors
+
+
+def test_list_as_source_referenced_before_created():
```

Comment:
As discussed, will need to cover the case when the list source is before the collector within the same section also

Reply:
Thanks, all seems to be added, there might be some very edge cases here and there but I think all is checked now.